filesize/filesize.go

```diff
@@ -0,0 +1,103 @@
+package filesize
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/NBISweden/sda-cli/download"
+	log "github.com/sirupsen/logrus"
+)
+
+// Help text and command line flags.
+
+// Usage text that will be displayed as command line help text when using the
+// `help list` command
+var Usage = `
+USAGE: %s filesize [url(s) | file]
+
+Filesize: List files that can be downloaded from the Sensitive Data Archive (SDA).
+	  If a directory is provided (ending with "/"), then the tool will attempt 
+	  to first download the urls_list.txt file, and then return a list of the
+	  files with their respective sizes.
+`
+
+// ArgHelp is the suffix text that will be displayed after the argument list in
+// the module help
+var ArgHelp = `
+  [url]
+        the first flagless argument will be used as file location.`
+
+// Args is a flagset that needs to be exported so that it can be written to the
+// main program help
+var Args = flag.NewFlagSet("filesize", flag.ExitOnError)
+
+// Function to return the size of a file
+func getFileSize(file string) (downloadSize int64, err error) {
+	resp, err := http.Head(file)
+	if err != nil {
+		return 0, fmt.Errorf("failed to head file, reason: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return 0, fmt.Errorf("failed to get file, code response not 200")
+	}
+
+	size, _ := strconv.Atoi(resp.Header.Get("Content-Length"))
+	downloadSize = int64(size)
+
+	return downloadSize, nil
+}
+
+// FileSize function returns the list of the files available for downloading and their
+// respective size. The argument can be a local file or a url to an S3 folder
+func FileSize(args []string) error {
+	// Parse flags. There are no flags at the moment, but in case some are added
+	// we check for them.
+	err := Args.Parse(args[1:])
+	if err != nil {
+		return fmt.Errorf("failed parsing arguments, reason: %v", err)
+	}
+
+	// Args() returns the non-flag arguments, which we assume are filenames.
+	urls := Args.Args()
+	if len(urls) == 0 {
+		return fmt.Errorf("failed to find location of files, no argument passed")
+	}
+
+	var currentPath, urlsFilePath string
+	currentPath, err = os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current path, reason: %v", err)
+	}
+
+	urlsFilePath, err = download.GetURLsListFile(currentPath, urls[0])
+	if err != nil {
+		return fmt.Errorf("failed to urls list file, reason: %v", err)
+	}
+
+	// Open urls_list.txt file and loop through file urls
+	urlsList, err := download.GetURLsFile(urlsFilePath)
+	if err != nil {
+		return err
+	}
```

Comment:
Try not have duplication in the different parts of the app.
This is handled by another argument in the application.

Reply:
I am not sure I follow this comment. Could you be a bit more specific?

Reply:
What I'm talking about is the fact that lot's of unintended stuff is happening by magic, in this case it is downloading a file when the expected action is to display the size of a file that can be downloaded.

Reply:
While we need the text file in order to get the filenames for all the other files there us no need for it to be saved to disk in this setting.

Reply:
The idea with saving the file (as it happens in the download module) is that I could have the same code for all the cases: when the user has the file already in their disk and when they pass a url to the file or to the containing folder. If that's not the case, there should be different code paths for files that exist in the disk and others that are supposed to be read with a request